editor/plugins/canvas_item_editor_plugin.cpp

```diff
@@ -183,11 +183,14 @@ bool CanvasItemEditor::_is_node_locked(const Node *p_node) {
 	return p_node->has_meta("_edit_lock_") && p_node->get_meta("_edit_lock_");
 }
 
-bool CanvasItemEditor::_is_node_movable(const Node *p_node) {
+bool CanvasItemEditor::_is_node_movable(const Node *p_node, bool p_popup_warning) {
 	if (_is_node_locked(p_node)) {
 		return false;
 	}
 	if (Object::cast_to<Control>(p_node) && Object::cast_to<Container>(p_node->get_parent())) {
+		if (p_popup_warning) {
+			_popup_warning_temporarily(warning_child_of_container, 3.0);
```

Comment:
It's a bit weird to pass a reference to a `Label` IMO, and it means that if we want other warnings we'll have to create labels for each of them. I think it's simpler to reuse the same design as for the warning dialog, where you popup the dialog directly with the arbitrary string you want to display (and thus reuse the same `Label` for all warnings).

Reply:
The problem is that I need the ability to check whether or not the warning is already displayed, so that it does not popup twice if it is already there. Using a string comparison is not really safe : as I want to expose this to plugins, they might not understand why their two warnings with the same text does do not popup.

So a solution could be to change the API to something like that:
```
int create_popup_warning("Warning: blablabla") // returns an ID
void popup_warning(int id, float time)
```

If you think this is cleaner I can change the API to that.

Reply:
Why not just use draw_string directly on the viewport, like in the other implementations?

Reply:
Well, depending on what you have behind it might be more or less easier to read (on white background for example). Also, this allow for several messages to be displayed, so we can reuse the system for other plugins. 
So in the end, it's easier using containers + labels instead of drawing a string.